doc/source/glossary.rst

```diff
@@ -0,0 +1,179 @@
+Glossary
+========
+There are many terms when talking about memory forensics, this list hopes to define the common ones and
+provide some commonality on how to refer to particular ideas within the field.
+
+A
+-
+.. _Address:
+    An address is another name for an :ref:`offset<Offset>`, specifically an offset within memory.  Offsets can be
+    both relative or absolute, whereas addresses are almost always absolute.
+
+.. _Address Space:
+
+Address Space
+    This is the name in volatility 2 for what's referred to as a :ref:`Translation Layer<translation layer>`.  It
+    encompasses all values that can be addresses, usually in reference to addresses in memory.
+
+.. _Alignment:
+
+Alignment
+    This value is what all data :ref:`offsets<offset>` will typically be a multiple of within a :ref:`type<type>`.
+
+.. _Array:
+
+Array
+    This represents a list of items, which can be access by an index, which is zero-based (meaning the first
+    element has index 0).  Items in arrays are almost always the same size (it is not a generic list, as in python)
+    even if they are :ref:`pointers<pointer>` to different sized objects.
+
+D
+-
+.. _Data Layer:
+
+Data Layer
+    A group of bytes, where each byte can be addressed by a specific offset.  Data layers are usually contiguous
+    chunks of data.
+
+.. _Dereference:
+
+Dereference
+    The act of taking the value of a pointer, and using it as an offset to another object, as a reference.
+
+.. _Domain:
+
+Domain
+    This the grouping for input values for a mapping or mathematical function.
+
+M
+-
+.. _Map:
+
+Map, mapping
+    A mapping is a relationship between two values, where one value (the :ref:`Domain<domain>` maps to the :ref:`Range<range>` value).
+    Mappings can be seen as a mathematical function, and therefore volatility 3 attempts to use mathematical functional
+    notation where possible.
+
+.. _Member:
+
+Member
+    The name of subcomponents of a type, similar to attributes of objects in common programming parlance.  These
+    are usually recorded as :ref:`offset<offset>` and :ref:`type<type>` pairs within a :ref:`structure<struct>`.
+
+O
+-
+.. _Object:
+
+Object
+    This has a specific meaning within computer programming (as in Object Oriented Programming), but within the world
+    of Volatility it is used to refer to a type that has been associated with a chunk of data.  See all :ref:`Type<type>`.
+
+.. _Offset:
+
+Offset
+    A numeric value that identifies a distance within a group of bytes, to uniquely identify a single byte, or the
+    start of a run of bytes.  This is often relative (offset from another object/item) but can be absolute (offset from
```

Comment:
Subjective: "This" seems confusing here.
I'd go with: "**An offset** is often relative (to another object / item)"

Reply:
I agree, updated.  5:)